docs/adr/0002-container-cve-remediation.md

```diff
@@ -0,0 +1,34 @@
+# 2. Container CVE remediation
+
+Date: 2025-02-13
+
+## Status
+
+Accepted
+
+## Context
+
+There is a stakeholder request for MintMaker to inform the users if any container updates fix known vulnerabilities. These pull requests should be appropriately marked and should be created at a higher frequency - ignoring general container updates schedules. This functionality enables the users to quickly integrate new vulnerability fixes and expedite the release process.
+
+## Decision
+
+Upstream renovate already has a similar feature, osvVulnerabilityAlerts, which gathers the vulnerabilities from OSV. Since OSV doesn't publish vulnerability data for containers, a custom solution had to be implemented. 
+
+Red Hat published advisories that describe which images are affected by the found vulnerabilities. This data cannot be used in a straightforward way, so I will describe the evaluation process of a vulnerability to a container image. The advisories contain the publication dates and a list of container images (among other artifacts) affected by the vulnerability. Since images are only uniquely identified by their digest, the list is non-exhaustive and only contains the newest image affected by a vulnerability in a given repo. To figure out if any random image is affected, its age relative to the mentioned image must be examined. In other words, images older than the mentioned one are affected; images newer than the mentioned one have fixed the vulnerability. In other ecosystems this can be accomplished by comparing versions of the packages. In the container ecosystem, the closest analog are tags. However, tags don't have a unified semantic meaning, and floating tags can change its underlying image over time. This means that it's impossible to determine if an image is affected by a vulnerability by comparing its tags. However, each image metadata contains the information of when an image was built, and advisories contain their publication date. With this information, we can claim that images older than an advisory are affected by the vulnerability, and images newer have fixed it. So, if renovate suggests an image bump pull request, if the old image is older and the new image is newer than a vulnerability that affects its repo, we can claim that the PR fixes a vulnerability.
```

Comment:
> images older than the mentioned one are affected

so the `affected` images in the osv database we generated are actually "fixed" because they're already have the vulnerability fixes installed, is it intentional to list the images under `affected` as a workaround or short-term solution? or it's a mistake?